### PR TITLE
basics operations running

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
Now the calculator performs basic operations. This commit does not yet contemplate operations with negative numbers, nor their concatenation or error handling

https://user-images.githubusercontent.com/73836200/105360860-d4ba2f80-5bd7-11eb-954f-c4e3ec6c2bbb.mp4

### CODE REVIEW

File: app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt [hunk 1]
```diff
@@ -32,11 +32,28 @@ class MainActivity : AppCompatActivity() {
...
 
     private fun numberPressed(button: Button) {
         button.setOnClickListener { presenter.numberPressed(button.text.toString()) }
     }
 
+    private fun operationSymbolPressed(button: Button) {
+        button.setOnClickListener { presenter.operationSymbolPressed(button.text.toString()) }
+    }
+
+    private fun operatorResultPressed(button: Button) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt]: if this is method is only called one time, I think is not necessary. You can put the listener directly in the listener method

File: app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt [hunk 2]
```diff
@@ -32,11 +32,28 @@ class MainActivity : AppCompatActivity() {
...
+        button.setOnClickListener { presenter.operatorResultPressed() }
+    }
+
+    private fun erase(button: Button) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt]: same here

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
+            firstOperand += number
+        } else {
+            secondOperand += number
+        }
     }
 
-    override fun getResult(): String {
-        return firstOperand
+    override fun getPartialResult(): String {
+        return (firstOperand + operator + secondOperand)
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Here is better if you do this 
override fun getPartialResult() = "$firstOperand$operator$secondOperand"
Its a better form when we concat strings

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt [hunk 4]
```diff
@@ -1,16 +1,59 @@
...
     }
+
+    override fun getFullResult(): String {
+        when (operator) {
+            OPERATOR_SUM -> result =
+                ((java.lang.Double.parseDouble(firstOperand) + java.lang.Double.parseDouble(
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: pls try to fix this import error

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt [hunk 5]
```diff
@@ -1,16 +1,59 @@
 package com.example.mykotlincalculator21.mvp.model
 
 import com.example.mykotlincalculator21.mvp.contract.CalculatorContract
-import com.example.mykotlincalculator21.utils.EMPTY_STRING
+import com.example.mykotlincalculator21.utils.*
```
Comment (lucianoeli) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: try to not use wild imports

Comment (lucianoeli) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: you can configure AS in File -> Settings
image

Comment (SartiMau) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: do not use wildcard imports

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 3 above)
Comment (SartiMau) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: the parenthesis here are not necessary

Comment (lucianoeli) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: you can ommit the return word too:

fun getPartialResult() = $firstOper...

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 4 above)
Comment (SartiMau) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: you can import java.lang.Double so here only call Double

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt [hunk 6]
```diff
@@ -1,16 +1,59 @@
...
+        when (operator) {
+            OPERATOR_SUM -> result =
+                ((java.lang.Double.parseDouble(firstOperand) + java.lang.Double.parseDouble(
+                    secondOperand
+                ))).toString()
+            OPERATOR_MINUS -> result =
+                ((java.lang.Double.parseDouble(firstOperand) - java.lang.Double.parseDouble(
+                    secondOperand
+                ))).toString()
+            OPERATOR_DIVIDE -> result =
```
Comment (SartiMau) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: why don't you check for the div by zero here? you will do this in a future PR?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 4 above)
Comment (lucianoeli) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: you could parse the operatos in a local variable, and then use it istead of parsing it in every case. its going to be more readable

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 6 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Yes that will be done in the next pr, in which i start to use all the others errors (enum class and all)

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Thx for the tip

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 3 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Thx for the tips guys

File: app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt (hunk 1 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt]: Your were right. now its clearer

File: app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt (hunk 2 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/MainActivity.kt]: Your were right. now its clearer

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Thx, i applied the solution that lucho gave me

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 3 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Thx, done it

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt (hunk 4 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Done it, thx guys

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt]: Done

File: app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt [hunk 7]
```diff
@@ -2,15 +2,51 @@ package com.example.mykotlincalculator21.mvp.model
...
 
-    override fun getResult(): String {
-        return firstOperand
+    override fun getPartialResult() = "$firstOperand$operator$secondOperand"
+
+    override fun getFullResult(): String {
+        var firstOperandValue = firstOperand.toDouble()
+        var secondOperandValue = secondOperand.toDouble()
+
+        return when (operator) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt:31]: Love it

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**